backend/src/interface/persistent_data.rs

```diff
@@ -26,77 +26,91 @@ pub enum DataError {
         additives: &[Additive],
-    ) -> Result<Option<Meal>>;
+    ) -> Result<Option<Uuid>>;
     /// Determines the side with the most similar name, identical allergens and identical additives.
+    /// Returns the uuid to the similar canteen.
     async fn get_similar_side(
         &self,
         similar_name: &str,
         allergens: &[Allergen],
         additives: &[Additive],
-    ) -> Result<Option<Side>>;
+    ) -> Result<Option<Uuid>>;
 
-    /// Updates an existing canteen entity in the database. Returns the entity.
-    async fn update_canteen(&self, uuid: Uuid, name: &str) -> Result<Canteen>;
-    /// Updates an existing line entity in the database. Returns the entity.
-    async fn update_line(&self, uuid: Uuid, name: &str) -> Result<Line>;
-    /// Updates an existing meal entity in the database. Returns the entity.
-    async fn update_meal(
+    /// Updates an existing canteen entity in the database.
+    /// Returns the canteen uuid.
+    async fn update_canteen(&self, uuid: Uuid, name: &str) -> Result<Uuid>;
+    /// Updates an existing line entity in the database.
+    /// Returns the line uuid.
+    async fn update_line(&self, uuid: Uuid, name: &str) -> Result<Uuid>;
+    /// Updates an existing meal entity in the database.
+    async fn update_meal(&self, uuid: Uuid, name: &str) -> Result<()>;
+    /// Adds a meal into the meal plan with the given params.
+    async fn add_meal_to_plan(
         &self,
-        uuid: Uuid,
-        line_id: Uuid,
+        canteen_id: Uuid,
         date: Date,
-        name: &str,
+        meal_id: Uuid,
         price: Price,
-    ) -> Result<Meal>;
-    /// Updates an existing side entity in the database. Returns the entity.
-    async fn update_side(
+    ) -> Result<()>;
+    /// Updates an existing side entity in the database.
+    async fn update_side(&self, uuid: Uuid, name: &str) -> Result<()>;
+    /// Adds a side into the meal plan with the given params.
+    async fn add_side_to_plan(
         &self,
-        uuid: Uuid,
-        line_id: Uuid,
+        canteen_id: Uuid,
         date: Date,
-        name: &str,
+        side_id: Uuid,
         price: Price,
-    ) -> Result<Side>;
+    ) -> Result<()>;
 
-    /// Adds a new canteen entity to the database. Returns the new entity.
-    async fn insert_canteen(&self, name: &str) -> Result<Canteen>;
-    /// Adds a new line entity to the database. Returns the new entity.
-    async fn insert_line(&self, name: &str) -> Result<Line>;
-    /// Adds a new meal entity to the database. Returns the new entity.
+    /// Adds a new canteen entity to the database.
+    /// Returns uuid of the new canteen.
+    async fn insert_canteen(&self, name: &str) -> Result<Uuid>;
+    /// Adds a new line entity to the database.
+    /// Returns uuid of the new line.
+    async fn insert_line(&self, name: &str) -> Result<Uuid>;
+    /// Adds a new meal entity to the database and creates a meal plan entry for the given day.
```

Comment:
#[allow(clippy::too_many_arguments)] allowed in this case?

Reply:
not neccesary